File: ruby/splitwise/lib/splitwise/models/user_balance.rb
module Splitwise
  module Models
    class UserBalance
      include Comparable

      attr_accessor :name, :balance

      def initialize(name:, balance:)
        self.name = name
        self.balance = balance
      end

      def <=>(other)
        balance <=> other.balance
      end
    end
  end
end

File: java/splitwise/src/main/java/gagan93jtg/lld/models/UserBalance.java
package gagan93jtg.lld.models;

import java.math.BigDecimal;

public class UserBalance {
  String name;
  BigDecimal balance;

  public UserBalance(String name, BigDecimal balance) {
    this.name = name;
    this.balance = balance;
  }

  public String getName() {
    return name;
  }

  public BigDecimal getBalance() {
    return balance;
  }

  public double getBalanceAsDouble() {
    return balance.doubleValue();
  }
}
